mitodl/ocw-hugo-themes
# feat: use youtube controls in videojs player

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1103 

#### What's this PR do?
This PR:

- Replaces our custom defined videoJs controller with YouTube controller (which enables YouTube keyboard shortcuts) for online courses.
- Applies the styling for video tabs (eg. Transcript, Related Resources) as shown in [link1](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347461/preview) and [link2](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347462/preview)
- Removes the extra space to the left of the interactive transcript, so it left aligns with the video player (see [InVision Comment](https://projects.invisionapp.com/d/#/console/21537330/472347462/comments/150602733))

#### How should this be manually tested?
This will have to be tested on such courses: **a)** Course that has interactive captions  **b)** Course that does not have interactive captions **c)** Course with Related resources **d)** Course with Lecture Notes
Reason for b) is that we want to verify that video tab bar still appears with the download popup.
The other a), c), d) are for verifying styling changes.
**Sample courses for testing:** 
1) [18.06-spring-2010](https://github.mit.edu/ocw-content-rc/18.06-spring-2010) (for interactive captions + related resources + download button)
2) [3.091-fall-2018](https://github.mit.edu/ocw-content-rc/3.091-fall-2018) (for only download button)
3) [15.s21-january-iap-2014](https://github.mit.edu/ocw-content-rc/15.s21-january-iap-2014) (for lecture notes)

For styling changes, check for mobile responsiveness as well.
**Steps to test:**

1. Clear your cache.
2. Use incognito on browser
3. Checkout to this branch.
Now repeat these steps for each course. (cache does not have to be cleared again)
4. `OCW_TEST_COURSE=course_id`
5. `yarn start course`
6. Go to http://localhost:3000/ and go to Video Lectures on the left panel. Open any video.
7. Verify that YouTube controller is playing the video as opposed to videoJs custom controller.
8. Make sure after you press Play, the keyboard shortcuts are functional (eg. F for full screen, arrow keys, c key for caption, etc)
9. Verify that the design/styling changes are according to InVision designs:  [link1](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347461/preview) and [link2](https://projects.invisionapp.com/d/main?origin=v7#/console/21537330/472347462/preview)


For testing non-interactive captions `3.091-fall-2018` course, verify that "Transcript" text does not appear but the download icon appears (see screenshots below). But to test it we will have to make some changes in our code, that's the only way to test it locally for now:

In the file: base-theme/layouts/partials/video.html
line 25-29:
`	{{ if $captionsLocation }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "transcript" "tabTitle" "Transcript" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{ else }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "" "tabTitle" "" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{end}}`

Replace this code with:
`	{{ if $captionsLocation }}
		{{ partial "video_expandable_tab.html" (dict "tabClass" "" "tabTitle" "" "downloadLink" $downloadLink "transcriptLink" $transcriptPdfLocation) }}
	{{end}}`

Save. This time do clear your cache, and run on incognito. Run `yarn start course`.  Make sure that the transcript button does not appear on video but the download icon still appears.

#### Any background context you want to provide?
It will be helpful for the reviewer to read this full discussion before testing this PR: https://github.com/mitodl/hq/discussions/418#discussion-4765581

For code review, these points are important for the design changes. The previous custom videoJs controller is not removed, but it is disabled by CSS like `display: none`. The reason for this is we may want to use it for offline courses sometime in the future.

#### Screenshots (if appropriate)

### For desktop:
<img width="897" alt="image" src="https://user-images.githubusercontent.com/109785089/228607777-30b39b8f-49bc-49e9-bc40-ef100cdeb56b.png">
<img width="843" alt="image" src="https://user-images.githubusercontent.com/109785089/228607892-4a2fd94e-c214-4348-a4d3-e5dea2e5d2ab.png">
<img width="900" alt="image" src="https://user-images.githubusercontent.com/109785089/228607979-c466965c-866b-4fb3-95f7-6430a5bd8c52.png">
<img width="837" alt="image" src="https://user-images.githubusercontent.com/109785089/228608792-a425a7cf-31ac-424c-98f9-486585cd0133.png">
<img width="855" alt="image" src="https://user-images.githubusercontent.com/109785089/228609212-b394b96d-7d2c-4d8b-b83f-1dfeb585e9df.png">
<img width="844" alt="image" src="https://user-images.githubusercontent.com/109785089/228609261-6c95b1ca-5219-4995-9c14-9f9e437de1fb.png">
<img width="1087" alt="image" src="https://user-images.githubusercontent.com/109785089/228609747-ec4eb615-e95e-416f-9163-c4ab6f90edc4.png">

### For mobile:
<img width="459" alt="image" src="https://user-images.githubusercontent.com/109785089/228611105-b5d989ea-8bea-400d-9f7c-cd431f63a579.png">
